### PR TITLE
Add organization

### PR DESCRIPTION
**Pull request by jgsqware**
I've updated the certificatetrp deps

https://github.com/giantswarm/giantswarm/issues/2062

### CODE REVIEW

File: service/resource/vaultcrt/create.go [hunk 1]
```diff
@@ -96,9 +96,8 @@ func (r *Resource) ensureVaultRole(customObject certificatetpr.CustomObject) err
 			AllowSubdomains:  AllowSubDomains,
 			AltNames:         key.AltNames(customObject),
 			ID:               key.ClusterID(customObject),
-			Organizations: []string{
-				key.ClusterComponent(customObject),
-			},
+			Organizations: append([]string{key.ClusterComponent(customObject)},
+				key.Organizations(customObject)...),
```
Comment (xh3b4sd) [inline service/resource/vaultcrt/create.go]: Above has to be done the same thing. It is not nice. There is an issue somewhere to make this more convenient.

File: service/key/key.go [hunk 2]
```diff
@@ -35,6 +35,10 @@ func IPSANs(customObject certificatetpr.CustomObject) []string {
 	return customObject.Spec.IPSANs
 }
 
+func Organizations(customObject certificatetpr.CustomObject) []string {
+	return customObject.Spec.Organizations
+}
```
Comment (xh3b4sd) [inline service/key/key.go]: Lets please do the append magic do in here. The organizations needed are the cluster component and the configured orgs in the tpo.

**Comment by jgsqware** [comment]
@xh3b4sd Is this better?

File: service/resource/vaultcrt/create.go [hunk 3]
```diff
@@ -111,15 +109,14 @@ func (r *Resource) ensureVaultRole(customObject certificatetpr.CustomObject) err
 }
 
 func (r *Resource) issueCertificate(customObject certificatetpr.CustomObject) (string, string, string, error) {
+
```
Comment (xh3b4sd) [inline service/resource/vaultcrt/create.go]: Why that newline?

File: service/resource/vaultcrt/create.go [hunk 4]
```diff
@@ -96,10 +96,8 @@ func (r *Resource) ensureVaultRole(customObject certificatetpr.CustomObject) err
 			AllowSubdomains:  AllowSubDomains,
 			AltNames:         key.AltNames(customObject),
 			ID:               key.ClusterID(customObject),
-			Organizations: []string{
-				key.ClusterComponent(customObject),
-			},
-			TTL: key.RoleTTL(customObject),
+			Organizations:    key.Organizations(customObject),
```
Comment (xh3b4sd) [inline service/resource/vaultcrt/create.go]: I like this ease, but as I said before, we also have to set the orgs above when checking if the role exists.

Comment (jgsqware) [reply service/resource/vaultcrt/create.go]: Good catch haven't seen it :)

**Pull request merged**